backend/src/mail/mail.service.ts

```diff
@@ -98,36 +100,35 @@ export class MailService {
         emcr_contact: '{{emcr_contact}}',
         bcws_contact: '{{bcws_contact}}',
         supervisor: '{{supervisor}}',
-        ...envs,
       }),
       attachments: [],
       contexts: records.map((record) => ({
         to: [
           templateType === TemplateType.MEMBER
-            ? record.personnel?.email
-            : record.personnel.supervisorEmail,
+            ? record?.personnel?.email
+            : record?.personnel?.supervisorEmail,
         ],
         cc: [],
         bcc: [],
         tag:
           templateType === TemplateType.MEMBER
-            ? `${template}_${record.personnel?.id}`
-            : `${template}_${record.personnel?.supervisorEmail}`,
+            ? `member`
+            : `supervisor`,
         delayTS: 0,
         context: {
           program: program?.toUpperCase(),
-          year: record.year,
+          year: `${record?.year}`,
           date: format(endDate, 'MMMM do, yyyy'),
-          member: `${record.personnel.firstName} ${record.personnel.lastName}`,
+          member: `${record?.personnel?.firstName} ${record?.personnel?.lastName}`,
           reason:
             template === EmailTemplates.MEMBER_DECLINED
-              ? record.supervisorReason
-              : record.memberReason,
+              ? `${record?.supervisorReason}`
+              : `${record?.memberReason}`,
           subject:
             templateType === TemplateType.SUPERVISOR
-              ? `${record.personnel.supervisorFirstName} ${record.personnel.supervisorLastName}`
-              : `${record.personnel.firstName} ${record.personnel.lastName}`,
-          supervisor: `${record.personnel.supervisorFirstName} ${record.personnel.supervisorLastName}`,
+              ? `${record?.personnel?.supervisorFirstName} ${record?.personnel?.supervisorLastName}`
```

Comment:
Are all these `?` necessary? For example, if this is a `records.map`, why would `record` ever be undefined?